modules/acquia_cms_common/src/EventSubscriber/KernelTerminate/AcquiaCmsTelemetry.php

```diff
@@ -254,13 +282,83 @@ private function shouldSendTelemetryData(): bool {
     if (AcquiaDrupalEnvironmentDetector::isAhEnv()) {
       $telemetryOpted = $this->state->get('acquia_connector.telemetry.opted', TRUE);
       $isAcquiaTelemetryDataSent = $this->state->get('acquia_cms_telemetry.status', FALSE);
-      if ($telemetryOpted && !$isAcquiaTelemetryDataSent) {
+      if ($telemetryOpted && !$isAcquiaTelemetryDataSent && $this->shouldResendData()) {
```

Comment:
Can we do something like this:
if (AcquiaDrupalEnvironmentDetector::isAhEnv() &&
$this->state->get('acquia_connector.telemetry.opted', TRUE) &&
$this->state->get('acquia_cms_telemetry.status', TRUE) &&
$this->shouldResendData()) {
    return TRUE;
}

Reply:
Same in this method line# 278-281 can be replace with this, check whether getenv("CI") is returning string or bool is string then do a typecasting otherwise not required.
if (getenv("CI")) {
      return FALSE;
    }